Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5120,8 +5131,8 @@ void WebPageProxy::setUserAgent(String&& userAgent, IsCustomUserAgent isCustomUs
 
     if (!hasRunningProcess())
         return;
-    forEachWebContentProcess([&](auto& webProcess, auto pageID) {
-        webProcess.send(Messages::WebPage::SetUserAgent(m_userAgent), pageID);
+    forEachWebContentProcess([userAgent = m_userAgent.isolatedCopy(), isCustomUserAgent](auto& webProcess, auto pageID) {
```

Comment:
There is no multi-threading involved here, why is this calling `isolatedCopy()` now?